runtime/vm/segment.c

```diff
@@ -880,3 +904,57 @@ printSegments(J9MemorySegment *s, void* data)
 	printf("\n");
 }
 #endif /* DEBUG_PRINT_SEGMENT_TYPES */
+
+
+BOOLEAN
+disclaimClassMemory(J9JavaVM *vm, UDATA flags)
+{
+	BOOLEAN rc = TRUE;
+#if defined(LINUX)
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9ClassLoaderWalkState walkState = {0};
+	J9ClassLoader *classLoader = vmFuncs->allClassLoadersStartDo(&walkState, vm, 0);
+	BOOLEAN enableROMClassDisclaim = J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags3, J9_EXTENDED_RUNTIME3_DISCLAIM_ROM_CLASS_MEMORY);
+	BOOLEAN enableRAMClassDisclaim = J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags3, J9_EXTENDED_RUNTIME3_DISCLAIM_RAM_CLASS_MEMORY);
+	BOOLEAN stopOnFirstFailure = J9_ARE_ANY_BITS_SET(flags, J9VM_DISCLAIMCLASSMEMORY_STOP_ON_FIRST_FAILURE);
+
+	while (NULL != classLoader) {
+		J9MemorySegment *segment = classLoader->classSegments;
+		while (NULL != segment) {
+			BOOLEAN isROM = J9_ARE_ANY_BITS_SET(segment->type, MEMORY_TYPE_ROM_CLASS);
+			BOOLEAN isRAM = J9_ARE_ANY_BITS_SET(segment->type, MEMORY_TYPE_RAM_CLASS);
+			if ((enableRAMClassDisclaim && isRAM && J9_ARE_ANY_BITS_SET(segment->type, MEMORY_TYPE_DISCLAIMABLE_TO_FILE))
+			|| (enableROMClassDisclaim && isROM)
```

Comment:
I see that disclaiming of ROM segments does not require `MEMORY_TYPE_DISCLAIMABLE_TO_FILE`. What is the backing storage in this case (assuming no SCC)?